src/dolphin/geckoCode.ts

```diff
@@ -135,7 +135,72 @@ export function saveCodes(iniFile: IniFile, codes: GeckoCode[]) {
     makeGeckoCode(code, lines);
   });
 
-  iniFile.setLines("Gecko", lines);
   iniFile.setLines("Gecko_Enabled", enabledLines);
   iniFile.setLines("Gecko_Disabled", disabledLines);
+  iniFile.setLines("Gecko", lines);
+}
+
+export function makeGeckoCodeFromRaw(rawGecko: string) {
+  const rawGeckoLines = rawGecko.split("\n");
+  let newCode: GeckoCode = {
+    name: "",
+    creator: "",
+    enabled: true,
+    defaultEnabled: false,
+    userDefined: true,
+    notes: [],
+    codeLines: [],
+  };
+
+  //fill out gecko info
+  rawGeckoLines.forEach((line) => {
```

Comment:
this switch case section is just reused from loadCodes so we should just make a new function to avoid duplication.